deploy/starter/infra/shared/sqlRoleAssignments.bicep

```
@@ -0,0 +1,32 @@
+@description('Principal ID to assign roles to.')
+param principalId string
+
+@description('Roles to assign.')
+param roleDefinitionIds object
+
+@description('CosmosDB Resource Id')
+param accountName string
+
+resource cosmosDb 'Microsoft.DocumentDB/databaseAccounts@2023-11-15' existing = {
+  name: accountName
+}
+
+/** Locals **/
+@description('Role Assignments to create')
+var roleAssignmentsToCreate = [for roleDefinitionId in items(roleDefinitionIds): {
+  name: guid(principalId, resourceGroup().id, roleDefinitionId.value)
+  roleDefinitionId: roleDefinitionId.value
+}]
+
+/** Resources **/
+resource roleAssignment 'Microsoft.DocumentDB/databaseAccounts/sqlRoleAssignments@2023-11-15' = [
+  for roleAssignmentToCreate in roleAssignmentsToCreate: {
+    name: guid(roleAssignmentToCreate.roleDefinitionId, principalId, cosmosDb.id)
+    parent: cosmosDb
+    properties: {
+      principalId: principalId
+      roleDefinitionId: '/${subscription().id}/resourceGroups/${resourceGroup().name}/providers/Microsoft.DocumentDB/databaseAccounts/${cosmosDb.name}/sqlRoleDefinitions/${roleAssignmentToCreate.roleDefinitionId}'

```

Use resource id function: https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#resourceid

Not tested:
```suggestion
       roleDefinitionId: resourceId('Microsoft.DocumentDB/databaseAccounts', cosmosDb.name, 'sqlRoleDefinition', roleAssignmentToCreate.roleDefinitionId)
```

---

I attempted to test after making this change and was met with the following error:

```
Deployment Error Details:
InvalidTemplate: Unable to process template language expressions for resource '/subscriptions/0a03d4f9-c6e4-4ee1-87fb-e2005d2c213d/resourceGroups/rg-fllm-0-4-0-aca/providers/Microsoft.DocumentDB/databaseAccounts/cosmos-pus65k7rkrjdw/sqlRoleAssignments/be05dc2f-03fd-521f-90fb-f2da08194b29' at line '1' and column '1223'. 'Unable to evaluate template language function 'resourceId': the type 'Microsoft.DocumentDB/databaseAccounts' requires '1' resource name argument(s). Please see https://aka.ms/arm-resource-functions/#resourceid for usage details.'
InvalidTemplate: Unable to process template language expressions for resource '/subscriptions/0a03d4f9-c6e4-4ee1-87fb-e2005d2c213d/resourceGroups/rg-fllm-0-4-0-aca/providers/Microsoft.DocumentDB/databaseAccounts/cosmos-pus65k7rkrjdw/sqlRoleAssignments/2462ae01-8939-5305-8247-e959ffa36fe9' at line '1' and column '1223'. 'Unable to evaluate template language function 'resourceId': the type 'Microsoft.DocumentDB/databaseAccounts' requires '1' resource name argument(s). Please see https://aka.ms/arm-resource-functions/#resourceid for usage details.'
```

I agree that this should be done the "right" way, but this isn't it.  Will investigate further after resolving other more pressing tasks.